### PR TITLE
explicit typecast unsigned

### PR DESCRIPTION
**Pull request by dakejahl**
This enforces type `unsigned` on the macros defined in `adc_interface.h`. I removed the negative in one of the defines, as it can just be applied at the line where the math is done. Since your ADC is outputting unsigned values, probably should keep all math in that type. Also cleaned up some functions for readability. This all saved a little bit of memory.

**Before**
```
   text	   data	    bss	    dec	
40304	    132	  11272	  51708
```
**After**
```
38592	    132	  11272	  49996
```

### CODE REVIEW

**Comment by dakejahl** [comment]
Rebased. Okay I'll stop pushing to this branch so you can merge.

File: Src/adc_interface.c [hunk 1]
```diff
@@ -99,7 +99,7 @@ uint8_t Set_Cell_One_Voltage(uint32_t adc_reading) {
 		return 0;
 	}
 
-	adc_values.cell_1_voltage = (adc_reading * CELL_ONE_ADC_SCALAR) + CELL_ONE_ADC_OFFSET;
+	adc_values.cell_1_voltage = (adc_reading * CELL_ONE_ADC_SCALAR) - CELL_ONE_ADC_OFFSET;
```
Comment (AlexKlimaj) [inline Src/adc_interface.c:102]: This starts getting confusing have some as addition and others as subtraction. I'd rather they all be addition in the math then have signed ints as #defines.

File: Src/adc_interface.c [hunk 2]
```diff
@@ -238,33 +238,30 @@ void HAL_ADC_ConvCpltCallback(ADC_HandleTypeDef* hadc) {
...
-		adc_filtered_output[0] = adc_buffer_filtered[0] / adc_sum_count;
-		adc_filtered_output[1] = adc_buffer_filtered[1] / adc_sum_count;
-		adc_filtered_output[2] = adc_buffer_filtered[2] / adc_sum_count;
-		adc_filtered_output[3] = adc_buffer_filtered[3] / adc_sum_count;
-		adc_filtered_output[4] = adc_buffer_filtered[4] / adc_sum_count;
+
+		length = sizeof(adc_filtered_output)/sizeof(adc_filtered_output[0]);
+
+		for(unsigned i = 0; i < length; i++) {
+			adc_filtered_output[i] /= adc_buffer[i];
```
Comment (AlexKlimaj) [inline Src/adc_interface.c:254]: /= typo

**Comment by AlexKlimaj** [comment]
Merged the adc_interface.c changes into master along with my changes to the #defines.

**Pull request closed without merging**